lib/reporting/reports/products_and_inventory/all_products.rb

```diff
@@ -19,7 +25,8 @@ def columns
           super.merge(
             {
               on_demand: proc{ |variant| variant.on_demand },
-              on_hand: proc{ |variant| variant.on_demand ? I18n.t(:on_demand) : variant.on_hand }
+              on_hand: proc{ |variant| variant.on_demand ? I18n.t(:on_demand) : variant.on_hand },
+              tax_category: proc { |variant| variant.tax_category_id && variant.tax_category.name }
```

Comment:
`tax_category` returns the default `TaxCategory` if the variant doesn't have one.
That's why we need to ensure that this tax category value maps to the variant's assigned tax category, rather than the default one.
This is the whole purpose of the addition of this field in the report.